### DISCUSSION TITLE
API and Forms

### DISCUSSION THREAD

**Discussion by IvanVaskovic** [symfony/symfony#46789] [Q&A] [answered]
Hello!

I'm using Forms to validate the incoming POST requests, and I have a rather interesting problem.

I've noticed that the form component casts all data types to `string` when `submit()` method is called. Normally, if you have a front-end where the form is rendered, this would be expected since all parameters would arrive as `strings` through `POST` method.

But since this is API, this is not desirable. For example, with `boolean` values, `true` gets cast to `"1"` and `false` is just an empty string which later gets interpreted as `NULL`.

Is there any way to stop the form component from automatically converting passed data to strings when `submit()` method is called?

**Comment by stof** [comment, marked as answer]
The Form component is designed to handle form submissions. So trying to (ab)use it for API requests will indeed likely cause issues (for instance, the CheckboxType expects the submission modal of HTML checkbox, i.e. omitting the value entirely if unchecked).

For APIs, using the Serializer component is generally a better idea (which is what is done in ApiPlatform for instance).

**Reply by IvanVaskovic** [reply]
Thank you Christophe for the answer, and I couldn't agree with you more, I've used the serializer component before and never had a problem.

However, in this particular situation, the whole project is already set up in this way. Code refactoring would last at least 2-3 months, and would cause a lot of headaches.

So, is there a way to disable automatic conversion to string without messing up the core code too much? 

I would really appreciate your help, thank you!

**Reply by stof** [reply]
I don't think there is one.
